### ISSUE TITLE
Chained Maps Are Handled Differently In v5 and v4

### ISSUE THREAD

**Issue by daniellking** [AutoMapper/AutoMapper#1610]
Hi,

I've noticed a serious discrepancy in how chained mappings are handled in v4 versus v5. Essentially, in v4 (specifically 4.2.1 when I tested) if a mapping returns null for an object, mappings further down the chain which would normally populate the object are not invoked. In v5, they appear to be invoked but with null as the source. This leads to a whole raft of problems, not the least of which is that every single property map would have to be replaced with a custom resolver to account for nulls. This currently prevents us from upgrading (even though we would like to take advantage of the performance improvements) because our domain to app logic relies heavily on this behavior due to our highly composited architecture. Almost all of our entities and DTOs are implementations of some combination of interfaces and we define the mappings largely by mapping "sister" interfaces from entity to the DTO layer. Then each real implementation of an entity to DTO map is merely a construction of `.IncludeBase`s for all the entities interfaces and the few properties unique to that entity. I have included a gist to demonstrate the problem (test passes in 4.2.1 and fails in 5.0.2) and would appreciate either an explanation for this breaking change or, preferably, a fix to return the original behavior.

https://gist.github.com/daniellking/462590d6a7a7f67b2f8409fc3d05d2c1

**Comment by jbogard** [comment]
Have you tried MyGet?

**Comment by daniellking** [comment]
No. I will try the latest dev version available and report back.

**Comment by daniellking** [comment]
Ok it works on 5.1.0-alpha-01199. Once that is merged into the main branch this issue should be resolved. I don't know if you want to add the test to the unit test suite or not (I'm not sure if this is a breakable regregression in your opinion). If you want me to update the gist for inclusion in some manner let me know and I'll be happy to do that.

**Comment by TylerCarlson1** [comment]
@daniellking There might already be a test for this, because you are the 4th-5th person who has had a null reference problem like this from the 5.x version.

**Comment by daniellking** [comment]
I'll look through the source and see if there is any test that covers this case. Though, to be clear, I never received a null reference exception (just a bunch of argument exceptions in code I wrote myself because arguments that shouldn't be passed as null were).

**Comment by daniellking** [comment]
Closing this since it is resolved in 5.1.1 (confirmed working in my codebase). I was not however able to get xunit running locally so I couldn't submit a pull request to add this unit test to the source :|
